### ISSUE TITLE
Handle different ways of function definitions in JS files

### ISSUE THREAD

**Issue by h3dkandi** [jeremyvii/vs-docblockr#31]
I am thankful for the functionality the extension currently have and I could live with it as it is but it would be nice to have this usability/comfort improvement.

**Is your feature request related to a problem? Please describe.**
If you define a function in a more weird way than the example provided in the Readme it will not work. Ex. code :

```
HS.Utils.usefullFunction = function usefullFunction(object) {
    return object
}
```

EDIT: Just found out that it doesn't work for anonymous functions as well (this might be categorised as bug?)
```
/**
function (par1, par2, par3){
    return "magic"
}
```
In this case it does not generate the parameters.
Combination of both two examples would be nice to work as well.

 I haven't research all the ways a function can be defined in JS.  I write only of what I have come across in my code for the little time I have used the extension.

**Describe the solution you'd like**
I expect when I write "//**" a top this line to work as it normally does.
Result should be:
```
/**
 * [usefullFunction description]
 *
 * @param   {[type]}  object  [object description]
 *
 * @return  {[type]}          [return description]
 */
HS.Utils.usefullFunction = function usefullFunction(object) {
    return object
}
```
**Describe alternatives you've considered**
Currently I have to move my cursor after the "=" hit two times enter in order to move the function definition to new line and use the "/**" atop of it, copy the generated template, undo those changes, paste the template above my definition. The code would look something like this before doing the "Undo"s:

```
HS.Utils.usefullFunction = 
/**
 * [usefullFunction description]
 *
 * @param   {[type]}  object  [object description]
 *
 * @return  {[type]}          [return description]
 */
function usefullFunction(object) {
    return object
}
```

**Comment by jeremyvii** [comment]
I agree, this is something that is needed. I will see about getting it implemented

**Comment by jeremyvii** [comment]
This should be fixed in the release [0.5.4](https://github.com/jeremyvii/vs-docblockr/releases/tag/0.5.4).
